sstables/mx/reader.cc

```diff
@@ -494,18 +494,20 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
                 _state = state::ROW_BODY_PREV_SIZE;
                 co_yield consumer_m::proceed::yes;
             }
-        row_body_prev_size_label:
           {
             // Ignore the result
             consumer_m::row_processing_result ret = _extended_flags.is_static()
                 ? _consumer.consume_static_row_start()
                 : _consumer.consume_row_start(_row_key);
 
-            if (ret == consumer_m::row_processing_result::retry_later) {
+            while (ret == consumer_m::row_processing_result::retry_later) {
```

Comment:
nitpick: if you used "do while" instead of "while", I think you wouldn't have had to duplicate the ret assignment code twice.

Reply:
I think I left the first ret assignment before the loop because if the comparison fails, we don't want to co_yield, so if we added the co_yield to a do_while loop, we would have to use an additional if anyway.

Also, we're using the ```ret``` value once again after the loop, so we needed to declare it before that anyway